src/docfx/build/metadata/Metadata.cs

```diff
@@ -22,5 +30,19 @@ where item.Match(file.FilePath)
 
             return JsonUtility.Merge(config.GlobalMetadata, fileMetadata);
         }
+
+        private static JObject GetGitRelated(Document file)
+        {
+            var result = new JObject();
+            var repo = GitRepoInfoProvider.GetGitRepoInfo(file);
+            if (repo == null)
+                return result;
+
+            var fullPath = Path.GetFullPath(Path.Combine(file.Docset.DocsetPath, file.FilePath));
+            var relPath = PathUtility.NormalizeFile(Path.GetRelativePath(repo.RootPath, fullPath));
+            result[MetadataConstants.GitCommit] = GitUtility.GetGitPermaLink(repo, relPath);
```

Comment:
>GitCommit [](start = 37, length = 9)

there is a case in PR that branch name is overwritten by input SourceBranchName

Reply:
add a TODO

---
In reply to: [194723137](https://github.com/dotnet/docfx/pull/2876#discussion_r194723137) [](ancestors = 194723137)